src/services/paymentTxOrMsgService.ts

```diff
@@ -0,0 +1,62 @@
+import { getPrismaClient } from '@helpers/database';
+import { v4 as uuidv4 } from 'uuid';
+import { generateRandomString } from '@helpers/generate-random-string';
+import {
+  Payload,
+  isEip681Payload,
+  isContractCallPayload,
+  isEip712Payload,
+} from '@/types/paymentTx';
+
+/**
+ * Helper for formatting the payload for storage in the database
+ * @returns
+ */
+export const createPaymentTxOrMsg = async (payload: Payload) => {
+  const prisma = getPrismaClient();
+
+  const paymentUuid = payload.uuid || uuidv4();
+  const verificationCode = generateRandomString();
+
+  const baseData = {
+    uuid: paymentUuid,
+    verificationCode,
+    chainId: payload.chainId,
+    dappUrl: payload.dappUrl,
+    dappName: payload.dappName,
+    payloadType: payload.payloadType,
+    
+    additionalPayload: payload.additionalPayload,
```

Comment:
Shouldn't this be thrown into `txParams` as well?

Reply:
I think I should store this one as it's own column, wdyt

Reply:
Hmm yeah I guess its own column makes sense